File: backend/app/schemas/audit.py
"""Lecture des trois journaux de traçabilité (spec.md § 6.1) — écriture déjà couverte
par `app.services.audit_trace_service`, ces schémas ne servent qu'à leur consultation
depuis l'IHM."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class FlowTraceRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    correlation_id: str
    direction: str
    afnor_api_version: str
    request: dict
    response: dict
    http_status: int
    created_at: datetime


class TechnicalLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    log_type: str
    origin: str
    company_id: int | None
    status: str
    new_count: int
    updated_count: int
    details: str | None
    created_at: datetime


class AuditLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    user_email: str | None = None
    action: str
    target: str
    ip_address: str | None
    created_at: datetime

File: frontend/src/api/audit.ts
import { apiFetch } from './http'

export interface FlowTrace {
  id: number
  correlation_id: string
  direction: string
  afnor_api_version: string
  request: unknown
  response: unknown
  request_headers: Record<string, string> | null
  response_headers: Record<string, string> | null
  http_status: number
  created_at: string
}

export interface TechnicalLog {
  id: number
  log_type: string
  origin: string
  company_id: number | null
  status: string
  new_count: number
  updated_count: number
  details: string | null
  created_at: string
}

export interface AuditLogEntry {
  id: number
  user_id: number | null
  user_email: string | null
  action: string
  target: string
  ip_address: string | null
  created_at: string
}

export function listFlowTraces(): Promise<FlowTrace[]> {
  return apiFetch('/api/ihm/audit/flow-traces', {}, 'Failed to list flow traces')
}

export function listTechnicalLogs(): Promise<TechnicalLog[]> {
  return apiFetch('/api/ihm/audit/technical-logs', {}, 'Failed to list technical logs')
}

export function listAuditLogs(): Promise<AuditLogEntry[]> {
  return apiFetch('/api/ihm/audit/audit-logs', {}, 'Failed to list audit logs')
}
